### PR TITLE
Fixes #23063 - Add notifications on subscription API errors

### PR DESCRIPTION
**Pull request by tstrachota**
I presume that we want to show notifications on errors at other pages too so I created a common helper that dispatches the error action and adds a notification. It also standardizes what data we send with API error actions.

### CODE REVIEW

File: webpack/move_to_foreman/common/helpers.js [hunk 1]
```diff
@@ -15,3 +17,17 @@ export const KEY_CODES = {
 };
 
 export const getResponseError = ({ data }) => data && (data.displayMessage || data.error);
+
+export const apiError = (actionType, result) => (dispatch) => {
+  const error = getResponseError(result.response);
+  dispatch({
+    type: actionType,
+    result,
+    error,
```
Comment (sharvit) [inline webpack/move_to_foreman/common/helpers.js]: Can you move the `result` and `error` to be under `payload` object?
I think we should always dispatch only `type` and `payload` and keep this as a standard.

**Comment by waldenraines** [comment]
I wonder if we want to remove the inline alert section now that we are showing toast notifications.  It seems a little redundant now:

screenshot from 2018-07-11 12-40-14

**Comment by waldenraines** [comment]
Is it possible that this also fixes [#24127](https://projects.theforeman.org/issues/24127)?

**Comment by tstrachota** [comment]
Updated:
- error actions now use format: `{ type, payload: {message}}`
- error alerts from the subscription page removed

**Comment by waldenraines** [comment]
> @waldenraines yes, it should fix that bug too

Do you want to add that to the commit message or maybe we can close [#23063](http://projects.theforeman.org/issues/23063) as a duplicate since [#24127](https://projects.theforeman.org/issues/24127) is tied to a redmine?

File: webpack/scenes/Subscriptions/Manifest/__tests__/manifest.fixtures.js [hunk 2]
```diff
@@ -210,7 +249,20 @@ export const deleteManifestFailureActions = [
...
+  },
+  {
+    payload: {
+      message: {
+        message: 'Request failed with status code 422',
+        sticky: true,
+        type: 'error',
+      },
+    },
+    type: 'TOASTS_ADD',
```
Comment (waldenraines) [inline webpack/scenes/Subscriptions/Manifest/__tests__/manifest.fixtures.js]: I know this is just a fixture but do we want to combine all of these into a `const` so we don't have to repeat ourselves here?

**Comment by tstrachota** [comment]
@waldenraines updated. I added test helpers for toast and failure actions. I also removed `api/fixtures.js` that's no longer used. It contained some repo request data but we're not importing it anywhere.

File: webpack/move_to_foreman/common/helpers.js [hunk 3]
```diff
@@ -1,11 +1,30 @@
...
+  `/${controller}/${id ? `${id}/` : ''}${action}`;
+
+const urlWithSearch = (base, searchQuery) =>
+  `/${base}?search=${searchQuery}`;
 
-  urlWithSearch(base, searchQuery) {
-    return `/${base}?search=${searchQuery}`;
-  },
+export const getResponseErrorMsg = ({ data }) =>
+  data && (data.displayMessage || data.message || data.error);
```
Comment (tstrachota) [inline webpack/move_to_foreman/common/helpers.js]: @waldenraines I saw your question about whether this handles list of errors in email (not sure why it disappeared from here):
Are there any responses that return **only** list of errors? From what I saw there's always a `displayMessage` with them and it seemed sufficient to display just that. We can check for list of errors and  send a notification for each of them. What do you think?

Comment (tstrachota) [reply webpack/move_to_foreman/common/helpers.js]: Per a discussion on irc:
We do need to process `errors` and they should take precedence over `displayMessage`.

**Comment by tstrachota** [comment]
Updated, `errors` in API responses now take precedence over `displayMessage`.

**Pull request merged**